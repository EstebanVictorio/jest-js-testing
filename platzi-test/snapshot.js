const getCharacter = ({ id, name, status, species, gender }) => ({
  id,
  name,
  status,
  species,
  gender
});

export { getCharacter };
