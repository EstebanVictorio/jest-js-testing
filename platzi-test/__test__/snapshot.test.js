import { getCharacter } from "../snapshot";
import rick from "../rick.json";

describe("Time for instant snapshots", () => {
  it("should snapshot", () => {
    expect(getCharacter(rick)).toMatchSnapshot();
  });

  it("should always fail", () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20)
    };

    expect(user).toMatchSnapshot();
  });

  it("should bring out an exception", () => {
    const user = {
      id: Math.floor(Math.random() * 20),
      name: "Steve Victory"
    };

    expect(user).toMatchSnapshot({ id: expect.any(Number) });
  });
});
