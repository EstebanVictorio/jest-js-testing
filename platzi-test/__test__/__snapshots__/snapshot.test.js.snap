// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time for instant snapshots should always fail 1`] = `
Object {
  "createdAt": 2019-08-26T20:22:05.298Z,
  "id": 6,
}
`;

exports[`Time for instant snapshots should bring out an exception 1`] = `
Object {
  "id": Any<Number>,
  "name": "Steve",
}
`;

exports[`Time for instant snapshots should snapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;
